### ISSUE TITLE
LoginSocialLinkedin for the first time not returning any data , everytime returning the second time,

### ISSUE THREAD

**Issue by udayan46** [cuongdevjs/reactjs-social-login#112]
Below is my code, whenever i click linkedinlogin on resolve getting error for the first time , second time its returning both provider and data
    <LoginSocialLinkedin
          
          client_id="****************"
          client_secret={"****************"}
          scope={"r_liteprofile r_emailaddress"}
          onLoginStart={this.linkedInLogin}
          redirect_uri={REDIRECT_URI}
          onResolve={({ provider, data }) => {
            this.onSocialLoginResolve1(provider, data);
          }}
          onReject={(err) => {
            console.log(err);
          }}>
          {/* <LinkedInLoginButton /> */}
          <div className="pt-4">
            <button type="button" className="btn-white">
              <svg
                className={classNames(
                  this.state.btnlinkedinLoader ? "" : "sr-only",
                  "animate-spin-medium h-5 w-5 rounded-full mx-2"
                )}
                viewBox="0 0 1024 1024"
                focusable="false"
                data-icon="loading"
                width="1em"
                height="1em"
                fill="currentColor"
                aria-hidden="true">
                <path d="M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 00-94.3-139.9 437.71 437.71 0 00-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3.1 19.9-16 36-35.9 36z"></path>
              </svg>
              <img className="pr-2" src="../../images/linkedin.png"></img>{" "}
              {LocaleStrings.mainlogin_signinlinkedin}
            </button>
          </div>
        </LoginSocialLinkedin>
        
        
        This is my on resolve function
        
        onSocialLoginResolve1 = (provider, data) => {
    console.log("provider", provider);
     console.log("data", data);

  };

**Comment by cuongdevjs** [comment]
It's really hard to detect where is the problem. I think the problem is your configure app. Have you faced the same problem when login on https://react-social-login.netlify.app?

**Comment by udayan46** [comment]
There is nothing much in the app,
Created a new app on developer's linkedin
added redirect url , one is the localhost:3000 and other is the main server
given access for sign in . Thats it

**Comment by cuongdevjs** [comment]
if you can, please reproduce a sample or give me more detail about the error so I can help you detect the problem. the Linkedin component is still working on the [site](https://react-social-login.netlify.app/)

**Comment by udayan46** [comment]
This is the error object i am getting

{
    "error": "invalid_request",
    "error_description": "Unable to retrieve access token: appid/redirect
uri/code verifier does not match authorization code. Or authorization code
expired. Or external member binding exists",
    "serviceErrorCode": 65600,
    "message": "Invalid access token",
    "status": 401
}

**Comment by udayan46** [comment]
As you can see here below first time nothing comes
the redirect uri is same both the places in the linkedin app and my reactjs
app i.e http://localhost:3000/
First time onResolve or onReject nothing is getting called second time from
that position it is working. Please suggest what i am doing wrong
this is my code below

<LoginSocialLinkedin
// isOnlyGetToken

client_id={CLIENT_ID}
client_secret={SECRET_KEY}
scope={"r_liteprofile r_emailaddress"}
onLoginStart={this.linkedInLogin}
redirect_uri={REDIRECT_URI}
onResolve={({ provider, data }) => {
console.log("data",data);
console.log("provider",provider)
}}
onReject={(err) => {
console.log("err", err);
}}>
<div className="pt-4">
<button type="button" className="btn-white">
<svg
className={classNames(
this.state.btnlinkedinLoader ? "" : "sr-only",
"animate-spin-medium h-5 w-5 rounded-full mx-2"
)}
viewBox="0 0 1024 1024"
focusable="false"
data-icon="loading"
width="1em"
height="1em"
fill="currentColor"
aria-hidden="true">
<path d="M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45
440.45 0 00-94.3-139.9 437.71 437.71 0 00-139.9-94.3C629 83.6 571.4 72 512
72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3
66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2
199.3.1 19.9-16 36-35.9 36z"></path>
</svg>
<img className="pr-2" src="../../images/linkedin.png"></img>{" "}
{LocaleStrings.mainlogin_signinlinkedin}
</button>
</div>
</LoginSocialLinkedin>

**Comment by cuongdevjs** [comment]
try uncomment isOnlyGetToken, you can send access token to server to get user's info.

**Comment by udayan46** [comment]
Same , please can you tell me why it is working for the second time ?
something with the version?
using this "reactjs-social-login": "^2.6.2",

**Comment by cuongdevjs** [comment]
no, I don't think so. The problem doesn't belong to this library, you can see it is working on my site. do onLoginStart called when you click on it the first time?

**Comment by udayan46** [comment]
Yes it's getting called

**Comment by cuongdevjs** [comment]
Next, a small window would open, do you see it or nothing happens?

**Comment by udayan46** [comment]
Yes the window opens and once i give the credentials for linekdin it closes
and comes the to redirect uri i.e http://localhost:3000/ but onResolve
nothing comes onReject nothing comes just stays there ,next time i don't
have to give any credentials as it takes the previous given credentials and
now onResolve it returns data and provider both

**Comment by cuongdevjs** [comment]
have you tried to open it with another browser? I can't guess what's problem.

**Comment by udayan46** [comment]
Yes same for all browsers

**Comment by cuongdevjs** [comment]
Maybe you should try to install another library

**Comment by udayan46** [comment]
Yes, working without any library .Thank you!

**Comment by rockers007** [comment]
@udayan46  : How do you fix this error? we are still facing the same issue.

**Comment by SouravBera-TE** [comment]
Hi, if anyone had already resolved this issue please let us know as we are still facing the same issue.
using this "reactjs-social-login": "^2.6.3" version.
